pkg/controller/bastion/options.go

```diff
@@ -163,30 +163,95 @@ func determineImageID(shoot *gardencorev1beta1.Shoot, providerConfig *awsv1alpha
 	return "", fmt.Errorf("found no suitable AMI for machines in region %q", shoot.Spec.Region)
 }
 
-func determineInstanceType(ctx context.Context, awsClient *awsclient.Client) (string, error) {
-	offerings, err := awsClient.EC2.DescribeInstanceTypeOfferingsWithContext(ctx, &ec2.DescribeInstanceTypeOfferingsInput{})
+func determineInstanceType(ctx context.Context, imageID string, awsClient *awsclient.Client) (string, error) {
+	var preferredType string
+	imageInfo, err := getImages(ctx, imageID, awsClient)
 	if err != nil {
-		return "", fmt.Errorf("failed to list instance types: %w", err)
+		return "", err
 	}
 
-	types := make([]string, len(offerings.InstanceTypeOfferings))
-	for i, offering := range offerings.InstanceTypeOfferings {
-		types[i] = *offering.InstanceType
+	imageArchitecture := imageInfo.Architecture
```

Comment:
do we need to check if nil or is it guaranteed that `Architecture` is never nil?

Reply:
committed  47b98430